File: src/cleansia_ios/CleansiaCustomer/Sources/Features/Disputes/PickedEvidence.swift
import Foundation

enum EvidenceUploadState: Equatable {
    case pending
    case uploading
    case uploaded
    case failed
}

/// A file the customer attached before the dispute exists: prepared on pick (compressed, validated,
/// written to a temp file) and uploaded to the dispute the create call returns.
struct PickedEvidence: Identifiable, Equatable {
    let id: String
    let fileName: String
    let file: PreparedEvidence
    var upload: EvidenceUploadState = .pending

    var isPdf: Bool {
        file.contentType.caseInsensitiveCompare("application/pdf") == .orderedSame
    }
}

File: src/cleansia_android/customer-app/src/main/java/cz/cleansia/customer/features/disputes/PickedEvidence.kt
package cz.cleansia.customer.features.disputes

sealed interface EvidenceUploadState {
    data object Pending : EvidenceUploadState
    data object Uploading : EvidenceUploadState
    data object Uploaded : EvidenceUploadState
    data object Failed : EvidenceUploadState
}

/**
 * A file the customer attached before the dispute exists. Held in memory until submit, when it is
 * uploaded to the dispute the create call just returned.
 *
 * A plain class, not a `data class`: a generated `equals` over [bytes] would compare identity.
 */
class PickedEvidence(
    val key: String,
    val fileName: String,
    val mimeType: String,
    val bytes: ByteArray,
    val upload: EvidenceUploadState = EvidenceUploadState.Pending,
) {
    val isPdf: Boolean get() = mimeType.equals("application/pdf", ignoreCase = true)

    fun withUpload(upload: EvidenceUploadState): PickedEvidence =
        PickedEvidence(key, fileName, mimeType, bytes, upload)
}
